iib/web/messaging.py

```diff
@@ -164,53 +238,46 @@ def send_message_for_state_change(request, new_batch_msg=False):
-    if request_address:
-        log.debug('Preparing to send a state change message for request %d', request.id)
-        properties = {
-            'batch': request_json['batch'],
-            'id': request_json['id'],
-            'state': request_json['state'],
-            'user': request_json['user'],
-        }
-        envelopes.append(json_to_envelope(request_address, request_json, properties))
-    else:
-        log.debug(
-            'No request state change message will be sent since the configuration '
-            '"IIB_MESSAGING_BUILD_STATE_DESTINATION" is not set'
-        )
+    request_envelope = _get_request_state_change_envelope(request)
+    if request_envelope:
+        envelopes.append(request_envelope)
 
-    batch_address = current_app.config.get('IIB_MESSAGING_BATCH_STATE_DESTINATION')
-    if batch_address:
-        if new_batch_msg:
-            # Avoid querying the database for the batch state since we know it's a new batch
-            batch_state = 'in_progress'
-        else:
-            batch_state = request.batch.state
+    batch_envelope = _get_batch_state_change_envelope(request.batch, new_batch_msg)
+    if batch_envelope:
+        envelopes.append(batch_envelope)
 
-        if new_batch_msg or batch_state in RequestStateMapping.get_final_states():
-            log.debug(
-                'Preparing to send a state change message for batch %d', request_json['batch']
-            )
-            content = {
-                'batch': request_json['batch'],
-                'request_ids': sorted(request.batch.request_ids),
-                'state': batch_state,
-                'user': request_json['user'],
-            }
-            properties = {
-                'batch': request.batch.id,
-                'state': batch_state,
-                'user': request_json['user'],
-            }
-            envelopes.append(json_to_envelope(batch_address, content, properties))
-    else:
-        log.debug(
-            'No batch state change message will be sent since the configuration '
-            '"IIB_MESSAGING_BATCH_STATE_DESTINATION" is not set'
-        )
+    if envelopes:
+        send_messages(envelopes)
+
+
+def send_messages_for_new_batch_of_requests(requests):
+    """
+    Send the appropriate message(s) based on a new batch of build requests.
+
+    If IIB is not configured to send messages, this function will do nothing.
+
+    :param list requests: the requests that were created as part of the batch request
+    """
+    if not requests:
+        return
+
+    envelopes = []
+
+    for request in requests:
+        request_envelope = _get_request_state_change_envelope(request)
+        if request_envelope:
+            envelopes.append(request_envelope)
+        else:
+            # If no request envelope was generated, then that means IIB is not configured to
+            # send request state change messages and the rest of the loop can be skipped
+            break
```

Comment:
This looks like a minor optimization that may cause some unexpected errors in the future due to unrelated code changes.